src/main/webapp/js/components/csv/csv-config.jsx

```diff
@@ -0,0 +1,80 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import 'react-table/react-table.css';
+
+import Select from '../../utils/inputs/select';
+import {
+  createCsvConfig,
+  deleteCsvConfig,
+  fetchCsvConfig,
+  saveCsvConfig,
+} from '../../actions';
+import {
+  SITE_ENTITY,
+  VACCINEE_ENTITY,
+  VISIT_ENTITY,
+} from '../../constants/entity-types';
+import CsvConfigPage from './csv-config-page';
+
+const ENTITY_TYPE = [
+  { label: 'Vaccinee', value: VACCINEE_ENTITY },
+  { label: 'Visit', value: VISIT_ENTITY },
+  { label: 'Site', value: SITE_ENTITY },
+];
+
+class CsvConfig extends Component {
```

Comment:
I think this should be split into two separate components, one for create (this one) and the other for update (and move getting the config from id there)
you will be able to pass the entire config to CsvConfigPage instead of the id